research/run_stats.py

```
@@ -0,0 +1,44 @@
+    self.scheduling_times.append(time_seconds)
+  
+  def set_expected_makespan(self, makespan: float) -> None:
+    self.expected_makespan = makespan
+
+  def set_transfer_start(self, data_item: int, size: float, time: float):
+    self.total_network_traffic += size;
+    self.transfer_starts[data_item] = time;
+
+  def set_task_start(self, task: int, resource: int, cores: int, memory: int, time: float):
+    self.current_cores += cores;
+    self.max_used_cores = self.max_used_cores.max(self.current_cores);
```

Ruff (0.15.12)

[error] Statement ends with an unnecessary semicolon

Remove unnecessary semicolon

(E703)